src/odemis/driver/piezomotor.py

```diff
@@ -819,26 +827,6 @@ def _sendCommand(self, cmd):
                     # don't send command again
                     raise IOError("Failed to read from PMT Control firmware, "
                                   "restarted serial connection.")
-
-            resp = b""
-            while resp[-len(EOL):] != EOL:
-                try:
-                    char = self._serial.read()
-                # TODO: what kind of exception is raised? Needs to be more specific.
-                except:
-                    logging.warning("Failed to read from PMT Control firmware, "
-                                    "trying to reconnect.")
-                    if self._recovering:
-                        raise
-                    else:
-                        self._tryRecover()
-                        # don't send command again
-                        raise IOError("Failed to read from PMT Control firmware, "
-                                      "restarted serial connection.")
-                if not char:
-                    raise IOError("Timeout after receiving %s" % to_str_escape(resp))
-                else:
-                    resp += char
```

Comment:
It seems that indeed `read_until()` can replace all the code above. However, you probably still want to keep this check. Now, instead of checking for an empty char, it should check whether the last byte is EOL. If not, that's due to a timeout => raise IOError